### PR TITLE
Feature/ignore custom route

### PR DESCRIPTION
**Pull request by Metalzoid**
Adding =

- initializer : config.ignored_paths = [] with docs
- configuration : @ignored_paths = [] && :ignored_paths
- route_extraction : 
- creating method ignore_custom_paths. Sanitize api_path, and check route with AND without api_path. Support controller#action AND controller (for ignoring all actions in this controller)

I hope the code is clean enough, don't hesitate to correct and explain to me if it doesn't suit you so that I can improve :)

### CODE REVIEW

**Comment by a-chacon** [comment]
Hey! Thank you! Again, I can review this at the end of the day. But at the first look I see it is ok. You can run rubocop in your local before upload the changes, so you can be sure you fix it ;)

**Comment by a-chacon** [comment]
And, please. Can you make a simple test for this feature? Like add a route to excluded paths, then check that your function returns what you expected? The file does not exist, so you should be create it. Something like /tests/lib/oas_rails/extractores/oas_route_extractor_test.rb. You can check the others files too.

**Comment by Metalzoid** [comment]
> And, please. Can you make a simple test for this feature? Like add a route to excluded paths, then check that your function returns what you expected? The file does not exist, so you should be create it. Something like /tests/lib/oas_rails/extractores/oas_route_extractor_test.rb. You can check the others files too.

I can try, but i never write a test ahah. I'm going how make that :dancing_men:

**Comment by a-chacon** [comment]
Don't worry, just look the others tests and you will know how to make it

**Comment by a-chacon** [comment]
mm I don't understand. If you show me what you try then I can help you. But I mean, you can do a simple test like this seudo code:

```ruby
      def test_ignored_paths
        OasRails.config.ignored_paths = ["usersController#index"]
        assert Extractors::RouteExtractor.ignore_custom_paths("/users")

        # then you can reset to default if needed
        OasRails.config.ignored_paths = []
      end
```

And try to test every branch in the method.

You can look (this)[https://github.com/a-chacon/oas_rails/blob/abef8d3abe772514b71c49aaa298ef1fe0c8aa37/test/lib/oas_rails/builders/responses_builder_test.rb#L13]

Is it what you can't do?

**Comment by Metalzoid** [comment]
It's a private method. I can use it ?

**Comment by a-chacon** [comment]
aa ok, I am sorry, I didn't see that. Then you can test it with the `host_routes` method, maybe? And use `clear_cache` method if it is needed for reset the variable?

**Comment by Metalzoid** [comment]
Hum ... see my test : 

```ruby
      def test_without_custom_routes
        OasRails.config.ignored_paths = []
        result = Extractors::RouteExtractor.host_routes
        assert_equal 14, result.count
        Extractors::RouteExtractor.clear_cache
        puts result.count
        result = []
        Configuration.new
        OasRails.config.ignored_paths = ["projects#index"]
        result = Extractors::RouteExtractor.host_routes
        puts result.count
        assert_equal 13, result.count
        Extractors::RouteExtractor.clear_cache
      end
``` 
I use Configuration.new for reinitialize configuration. It's work with ["projects#show"], but don't work with ["projects"], or, in my app, it's works fine :sob:

**Comment by Metalzoid** [comment]
it's all done :love_you_gesture:

File: CHANGELOG.md
Comment (a-chacon) [inline CHANGELOG.md:1]: hey! This file is created automatically. There is no need for touch it. I think you open it and your code editor formatted it?

File: test/lib/oas_rails/extractors/route_extractor_test.rb
Comment (a-chacon) [inline test/lib/oas_rails/extractors/route_extractor_test.rb:1]: Here I see three tests. Can you separate it with more specific names please? like test_without_custom_routes,then test_with_custom_routes_controller_action and then test_with_custom_routes_controller

**Review by a-chacon — COMMENTED** [review]
After resolve this, we are ok with your feature! Thanks again!

File: CHANGELOG.md
Comment (Metalzoid) [reply CHANGELOG.md:1]: i don't touch that, idk?? I restore from main that file

**Review by a-chacon — APPROVED** [review]

**Comment by a-chacon** [comment]
hey @Metalzoid thank you! I will release a new version at the night.

**Comment by a-chacon** [comment]
We have problems :( Why it fails in main and not in this tests? 

```
Run options: --seed 56473
# Running:
.........................F
Failure:
OasRails::Builders::RouteExtractorTest#test_with_custom_controllers_actions [test/lib/oas_rails/extractors/route_extractor_test.rb:14]:
Expected: 13
  Actual: 14
bin/rails test test/lib/oas_rails/extractors/route_extractor_test.rb:11
F
Failure:
OasRails::Builders::RouteExtractorTest#test_with_custom_controller_only [test/lib/oas_rails/extractors/route_extractor_test.rb:21]:
Expected: 8
  Actual: 14
bin/rails test test/lib/oas_rails/extractors/route_extractor_test.rb:18
.
Finished in 0.067122s, 417.1500 runs/s, 849.1981 assertions/s.
28 runs, 57 assertions, 2 failures, 0 errors, 0 skips
```

Do you know what could be happening here?

https://github.com/a-chacon/oas_rails/actions/runs/10620740013/job/29441251829?pr=53

**Comment by Metalzoid** [comment]
Oh yes, I see. M'y bad ! The test works with test environment (route you are created on the test environment). The result is write in brut, not auto calculated. 
I try to fix it tomorrow ok ?

**Comment by a-chacon** [comment]
yes! don't worry. We leave the release for the weekend. I really appreciate your interest in helping with this.

About the failing tests. Mm give it a look. All the tests run with the test environment, so could be another thing.

**Pull request merged**